bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java

```diff
@@ -0,0 +1,447 @@
+ * querying a Website/Device.
+ * 
+ * @author Laurens Van Acker
+ * @since 1.8.0
+ */
+public class SonanceBinding extends
+		AbstractActiveBinding<SonanceBindingProvider> {
+	private Map<String, Socket> socketCache = new HashMap<String, Socket>();
+	private Map<String, DataOutputStream> outputStreamCache = new HashMap<String, DataOutputStream>();
+	private Map<String, BufferedReader> bufferedReaderCache = new HashMap<String, BufferedReader>();
+	private Map<String, ReentrantLock> volumeLocks = new HashMap<String, ReentrantLock>();
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(SonanceBinding.class);
+
+	/**
+	 * the refresh interval which is used to poll values from the Sonance server
+	 * (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	public SonanceBinding() {
+	}
+
+	/**
+	 * Called by the SCR to activate the component with its configuration read
+	 * from CAS
+	 * 
+	 * @param bundleContext
+	 *            BundleContext of the Bundle that defines this component
+	 * @param configuration
+	 *            Configuration properties for this component obtained from the
+	 *            ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext,
+			final Map<String, Object> configuration) {
+		// the configuration is guaranteed not to be null, because the component
+		// definition has the
+		// configuration-policy set to require. If set to 'optional' then the
+		// configuration may be null
+
+		// to override the default refresh interval one has to add a
+		// parameter to openhab.cfg like <bindingName>:refresh=<intervalInMs>
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString))
+			refreshInterval = Long.parseLong(refreshIntervalString);
+
+		setProperlyConfigured(true);
+	}
+
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed
+	 * through the ConfigAdmin service.
+	 * 
+	 * @param configuration
+	 *            Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString))
+			refreshInterval = Long.parseLong(refreshIntervalString);
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the
+	 * configuration is removed or mandatory references are no longer satisfied
+	 * or the component has simply been stopped.
+	 * 
+	 * @param reason
+	 *            Reason code for the deactivation:<br>
+	 *            <ul>
+	 *            <li>0 – Unspecified
+	 *            <li>1 – The component was disabled
+	 *            <li>2 – A reference became unsatisfied
+	 *            <li>3 – A configuration was changed
+	 *            <li>4 – A configuration was deleted
+	 *            <li>5 – The component was disposed
+	 *            <li>6 – The bundle was stopped
+	 *            </ul>
+	 */
```

Comment:
remove generic javadoc and explain what YOUR version of this method is going to do.